### DISCUSSION TITLE
Make exclude_unset work in pydantic models dependencies in form body

### DISCUSSION THREAD

**Discussion by drforse** [fastapi/fastapi#8384] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, Form, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    x: str = Form()


@app.get("/")
async def foo(item: Item = Depends()):
    print(item.dict(exclude_unset=True))
```

### Description

Open the browser and call endpoint "/"
It prints `{}`

### Wanted Solution

I would like it not to fill dependency model with nulls, but leave it unset, with a parameter in Depends for backward compatibilty

### Wanted Code

```python
from fastapi import Depends, Form, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    x: str = Form()


@app.get("/")
async def foo(item: Item = Depends(leave_unset=True)):
    print(item.dict(exclude_unset=True))
```

### Alternatives

Make Form() in fields in models work without Depends() in handler function. Right now it is considered as application/json

### Operating System

Windows

### Operating System Details

Windows 11

### FastAPI Version

0.78.0

### Python Version

3.10.6

**Comment by csrgxtu** [comment]
are you want to access `http://localhost:8000/` with parameter unset?

**Comment by drforse** [comment]
> are you want to access `http://localhost:8000/` with parameter unset?

I want to access it with parameter unset and in this case the print be {} instead of {"x": None} as it is now

**Comment by JarroVGIT** [comment]
> Open the browser and call endpoint "/"
It prints {}

And

> I want to access it with parameter unset and in this case the print be {} instead of {"x": None} as it is now


Seem to be the opposite of each other..

**Comment by drforse** [comment]
> > Open the browser and call endpoint "/"
> > It prints {}
> 
> And
> 
> > I want to access it with parameter unset and in this case the print be {} instead of {"x": None} as it is now
> 
> Seem to be the opposite of each other..

Really? May be I don't construct my sentences correctly. It seems perfectly same to me...
What I mean is: now it prints {"x": None}, but I want it to print {} (because exclude_unset is True and "x" wasn't passed in request)

**Comment by JarroVGIT** [comment]
I see. You could use `exclude_none=True` like so:

```python
class Item(BaseModel):
    x: str | None = Form()


@app.get("/")
async def foo(item: Item = Depends()):
    print(item.dict(exclude_unset=True, exclude_none=True))
```
This would print `{}` as the default value is `None`. Not exactly the same you were trying to achieve but maybe might work in your specific use case? It would mean that you would not allow `null` values as a valid input in certain `Form()` fields.

**Comment by drforse** [comment]
> I see. You could use `exclude_none=True` like so:
> 
> ```python
> class Item(BaseModel):
>     x: str | None = Form()
> 
> 
> @app.get("/")
> async def foo(item: Item = Depends()):
>     print(item.dict(exclude_unset=True, exclude_none=True))
> ```
> 
> This would print `{}` as the default value is `None`. Not exactly the same you were trying to achieve but maybe might work in your specific use case? It would mean that you would not allow `null` values as a valid input in certain `Form()` fields.

I know about exclude_none, but that's not exactly what I wanted. That's different things

**Comment by Gipssik** [comment]
It looks like I have similar problem, so...

I have this schema:
````
class GetStatisticsQueries(BaseModel):
    match_id: int | None = None
    game_id: int | None = None
    statistics_type: StatisticsType | None = None
````
And I want to use it as query parameters schema this way:
```
@router.get(
    '',
    summary='Get statistics by the given fields',
    response_model=list[StatisticsSchema],
)
async def get_statistics(
        queries: GetStatisticsQueries = Depends(),
        ...
) -> list[dict]:
    queries_dict = queries.dict(exclude_unset=True)
```
The problem is that `dict(exclude_unset=True)` returns all values every time even if value was not set. I guess it happens because of `Depends()` call.

**Comment by 9en9i** [comment]
Faced the same problem. @drforse and @Gipssik do you have any solutions?

**Reply by drforse** [reply]
There is no solution really, the issue was created not to find a solution but to propose a change in logic.
Right now the only option is to set a default value None and use exclude_none which is not the same and I don't like it... Well, you can use some custom solutions, like make a default value an object of a class and create a function that excludes what you want, but that's already on your side of code, or you can customize the library code in any way for yourself... So the point is, there is no built-in solution besides exclude_none

**Comment by RafaelWO** [comment]
I have a similar problem, although I don't use `Form`.

```python
import uvicorn
from fastapi import Depends, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    x: str | None
    y: int | None


@app.get("/")
async def foo(item: Item = Depends()):
    return item.dict(exclude_unset=True)


@app.post("/")
async def bar(item: Item):
    return item.dict(exclude_unset=True)


if __name__ == '__main__':
    uvicorn.run(app)
```

When you POST an empty body `{}` to `/` it returns and empty dict `{}` as expected because `exclude_unset=True` works.

When you do a GET `/` without any query params, I would expect the same. However it returns a dict with all keys set to `null`.

```json
{
  "x": null,
  "y": null
}
```

I.e. `exclude_unset=True` does not work when using `Depends()` to construct query parameters via a pydantic model.

**Comment by cfytrok** [comment]
The problem is that this line sets a default value https://github.com/fastapi/fastapi/blob/2809c08a70c5b429a8fe803a5d62802d44d066be/fastapi/dependencies/utils.py#L736
